### PR TITLE
Allow empty value for config set

### PR DESCRIPTION
**Pull request by chandrareddyp**
### What this PR does / why we need it
This PR is to make sure the `<value>` for the `tanzu config set <key> <value>` is not mandatory, the `<value>` can be empty string.
### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR
Unit test cases added.

### Release note

```release-note

```

### CODE REVIEW

File: config/envs.go [hunk 1]
```diff
@@ -42,11 +42,6 @@ func GetEnv(key string) (string, error) {
 }
 
 func getEnv(node *yaml.Node, key string) (string, error) {
-	// check if key is empty
```
Comment (mpanchajanya) [inline config/envs.go:45]: @chandrareddyp we need this validation. we dont want key to be empty, only value can be empty

File: config/envs.go [hunk 2]
```diff
@@ -133,11 +128,6 @@ func SetEnv(key, value string) (err error) {
 
 //nolint:dupl
 func setEnv(node *yaml.Node, key, value string) (persist bool, err error) {
-	// check if key is empty
```
Comment (mpanchajanya) [inline config/envs.go:136]: @chandrareddyp Same here as well. 

we don't want key to be empty, only value can be empty

**Review by mpanchajanya — CHANGES_REQUESTED** [review]
key cannot be empty. only value can be empty, Try removing validation on value and update the tests as needed

File: config/envs.go (hunk 1 above)
Comment (chandrareddyp) [reply config/envs.go]: Yes, i raised PR in a rush, i have updated.

**Review by mpanchajanya — APPROVED** [review]
lgtm

**Pull request merged**